packages/core/src/components/Modal/useShowHideModal.ts

```diff
@@ -37,22 +38,28 @@ export default function useShowHideModal({
     keys: KEYS
   });
 
-  // show/hide and animate the modal
   useEffect(() => {
-    if (!instance) {
-      return;
-    }
-    const animate = instance?.$el.childNodes[1].animate;
     if (show) {
+      setIsAnimating(true);
       instance?.show();
-      if (animate) instance?.$el.childNodes[1].animate?.(...getAnimationProps());
+      const animate = instance?.$el.childNodes[1].animate;
+      if (animate) {
+        instance?.$el.childNodes[1].animate?.(...getAnimationProps());
+      }
     } else {
+      const animate = instance?.$el.childNodes[1].animate;
       if (animate) {
         const animation = instance?.$el.childNodes[1]?.animate(...getAnimationProps(true));
-        animation?.finished.then(() => instance?.hide());
+        animation.finished.then(() => {
```

Comment:
why did you removed the "?" after "animation"?

Reply:
Missed it, fixing